spin/cmds/meson.py

```diff
@@ -130,6 +137,18 @@ def build(meson_args, jobs=None, clean=False, verbose=False, quiet=False):
     """
     build_dir = "build"
     setup_cmd = _meson_cli() + ["setup", build_dir, "--prefix=/usr"] + list(meson_args)
+    env = os.environ
+
+    if gcov_build:
+        # Ref: https://github.com/numpy/numpy/pull/3979/files
```

Comment:
I'm not sure if I'm doing something wrong here, the `.gcda` generated from old `runtest.py` is of the format:
```
./build/temp.linux-x86_64-3.10/build/src.linux-x86_64-3.10/numpy/core/src/umath/matmul.gcda
```

while the new one looks like:
```
./build/numpy/_core/_multiarray_umath.cpython-310-x86_64-linux-gnu.so.p/meson-generated_matmul.c.gcda
```

This means `lcov` is not working as expected. I'll take a look at this tomorrow, something off with the flags I'm guessing.

Reply:
Meson supports coverage https://mesonbuild.com/howtox.html#producing-a-coverage-report, making life a lot easier.